tests/nn/data_parallel/test_gossip.py

```diff
@@ -0,0 +1,778 @@
+    }
+
+    mp.spawn(
+        run_test_slowmo_with_slowmo_freq_1,
+        args=(world_size, temp_file_name, slowmo_settings),
+        nprocs=world_size,
+        join=True,
+    )
+
+
+@requires_nccl()
+@skip_if_single_gpu
+def test_nccl_backend_2_proc_1_node() -> None:
+    world_size = 2
+    temp_file_name = tempfile.mkstemp()[1]
+    slowmo_settings = {
+        "slowmo_base_algorithm": gossip.SlowmoBaseAlgorithm.LOCALSGD,
+        "localsgd_frequency": 100,  # Localsgd has to be disabled since it would fail in the 1 node case. TODO: Need to allow it to run without failing in GossipDataParallel in the one node case
+        "nprocs_per_node": 2,
+        "slowmo_momentum": 0.0,
+    }
+
+    mp.spawn(
+        run_test_slowmo_with_slowmo_freq_1,
+        args=(world_size, temp_file_name, slowmo_settings),
+        nprocs=world_size,
+        join=True,
+    )
+
+
+# @requires_nccl()
+# @skip_if_lt_x_gpu(4)
+# def test_nccl_backend_2_proc_2_node():
+#     # 2 device, 2 node
+#     # 4 device, 1 node
+#     # 1 device, 4 node
+#     # can change world size to 4
+#     # will need to change world_size to 4 for this
+#     world_size = 4
+#     temp_file_name = tempfile.mkstemp()[1]
+#     slowmo_settings = {
+#         "slowmo_base_algorithm": gossip.SlowmoBaseAlgorithm.LOCALSGD,
+#         "localsgd_frequency": 1,
+#         "rank": rank,
+#         "world_size": world_size,
+#         "nprocs_per_node": 2,
+#         "local_node_group": process_group,
+#         "master_group": process_group,
+#         "slowmo_momentum": 0.0,
+#     }
+
+#     mp.spawn(
+#         run_test_slowmo_with_process_group,
+#         args=(world_size, temp_file_name, process_group, slowmo_settings),
+#         nprocs=world_size,
+#         join=True,
+#     )
+
+
+# FIXME: @lefaudeux - most of the tests below could be factorized and semi-autogenerated
+
+
+@requires_nccl()
+@skip_if_single_gpu
+def test_localsgd_slowmo_freq_1() -> None:
+    world_size = 2
+    temp_file_name = tempfile.mkstemp()[1]
+    slowmo_settings = {
+        "slowmo_base_algorithm": gossip.SlowmoBaseAlgorithm.LOCALSGD,
+        "localsgd_frequency": 1,
+        "nprocs_per_node": 1,
+        "slowmo_momentum": 0.5,
+        "slowmo_frequency": 1,
+        "slowmo_memory_efficient": True,
+    }
+
+    model_optimizer_momentum = 0.5
+
+    mp.spawn(
+        run_test_slowmo_with_slowmo_freq_1,
+        args=(world_size, temp_file_name, slowmo_settings, model_optimizer_momentum,),
+        nprocs=world_size,
+        join=True,
+    )
+
+
+@requires_nccl()
+@skip_if_single_gpu
+def test_sgp_slowmo_freq_1() -> None:
+    world_size = 2
+    temp_file_name = tempfile.mkstemp()[1]
+
+    model_optimizer_momentum = 0.5
+    slowmo_settings = {
+        "slowmo_base_algorithm": gossip.SlowmoBaseAlgorithm.SGP,
+        "nprocs_per_node": 1,
+        "slowmo_momentum": 0.5,
+        "slowmo_frequency": 1,
+        "slowmo_memory_efficient": False,
+    }
+
+    mp.spawn(
```

Comment:
@vtantia here the optimizer momentum is not actually being used, "run_test_slowmo_with_slowmo_freq_1" does not use this input

Reply:
That's a good catch thanks! I'll delete it from the function. I was earlier keeping both this and the slowmo_momentum in slowmo_settings the same, but later decided to switch to using the latter value, and forgot to remove this